src/legacy/ui/public/vis/editors/config/editor_config_providers.ts

```diff
@@ -128,37 +128,26 @@ class EditorConfigProviderRegistry {
     current: TimeIntervalParam,
     merged: EditorParamConfig,
     paramName: string
-  ): { timeBase?: string; default?: string } {
+  ): { timeBase: string; default: string } {
```

Comment:
:information_source:  The problem here was, that the method could from it's types return an object without `default`, that would then legibly fail (in newer TS versions) later on, because TS detects correctly, that a `TimeIntervalParam` needs this, so this merge method must return it. In fact it also does, if we remove the last `if` check for `isTimeBaseParam(current)` which is anyway redundant, since we know that `current` is a `TimeIntervalParam` (it's the type of the parameter).